extensions/elytron-security-oauth2/deployment/src/main/java/io/quarkus/elytron/security/oath2/deployment/OAuth2DeploymentProcessor.java

```diff
@@ -0,0 +1,138 @@
+package io.quarkus.elytron.security.oath2.deployment;
+
+import org.jboss.logging.Logger;
+import org.wildfly.security.auth.server.SecurityRealm;
+
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.substrate.SubstrateResourceBuildItem;
+import io.quarkus.elytron.security.deployment.*;
+import io.quarkus.elytron.security.oauth2.runtime.OAuth2Config;
+import io.quarkus.elytron.security.oauth2.runtime.OAuth2Template;
+import io.quarkus.elytron.security.runtime.*;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.undertow.deployment.ServletExtensionBuildItem;
+import io.undertow.security.idm.IdentityManager;
+import io.undertow.servlet.ServletExtension;
+
+/**
+ * The build time process for the OAUth2 security aspects of the deployment. This creates {@linkplain BuildStep}s for
+ * integration
+ * with the Elytron OAUth2 security services. This supports the Elytron OAuth2
+ * {@linkplain org.wildfly.security.auth.realm.token.TokenSecurityRealm realm implementations.
+ */
+class OAuth2DeploymentProcessor {
+    private static final Logger log = Logger.getLogger(OAuth2DeploymentProcessor.class.getName());
+    private static final String REALM_NAME = "Oauth2";
+    private static final String AUTH_MECHANISM = "BEARER_TOKEN";
+
+    OAuth2Config oauth2;
+
+    /**
+     * Register the CDI beans that are needed by the OAUth2 extension
+     *
+     * @param additionalBeans - producer for additional bean items
+     */
+    //FIXME javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type org.wildfly.security.auth.realm.token.validator.OAuth2IntrospectValidator$1 and qualifiers [@Default]
```

Comment:
is this commented step needed ?

Reply:
I didn't tested the native mode yet, there should be some stuff to be done on this side. So two options here:
1. Accept the PR and open an issue to check the extention in native
2. Check it in native before mergin: so the commented step stays as a reminded

Reply:
@loicmathieu if it is not an immediate issue for you then I guess the native mode can be supported at the next stage, but it is only my opinion. I'm not sure how strict the requirements are to the new extensions about the native mode

Reply:
@sberyozkin we do aim to have all extensions support both HotSpot and native. This is a core contract of Quarkus. I'd rather we don't deviate from that without good reason.

Reply:
@emmanuelbernard sure, @loicmathieu, FYI

Reply:
Hello @emmanuelbernard,
Native mode is working but not when you use a custom cert for the introspection endpoint (the SSLContext related stuff is not working in native).
If it's OK, we can merge the PR and open a followup issue for this.
Else, I will need to understand what's going on ...

Reply:
@loicmathieu, maybe this can help with the SSLContext issue when doing native https://github.com/quarkusio/quarkus/blob/c9cba824e8812fa3f15474b8382ac5d90f7238aa/extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakDeploymentProcessor.java#L22.

Reply:
@pedroigor this doesn't solve the issue. Maybe I'm doing something that is not possible with SVM ... I allow a user to register a custom certificate, and so to use a custom SSLContext, this seems to be impossible with SVM so maybe this functionality needs to be restricted from native that should use the standard way : integrate the certificate in the JVM cert file and provide a dedicated cert file at runtime.